torchtitan/config_manager.py

```diff
@@ -560,6 +559,12 @@ def __init__(self):
             type=Optional[str],
             default=None,
         )
+        self.parser.add_argument(
+            "--logging.log_level",
+            type=test_fail_val,
```

Comment:
what is test_fail_val? 
I couldn't find its explaination.
Maybe also add  default="INFO" field?

Reply:
By the way, wouldn’t it be simpler to just add the restriction using only argparse? Perhaps something like this:

self.parser.add_argument(
            "--logging.log_level",
            type=str,
            default="INFO",
            choices=["INFO", "DEBUG", "WARNING", "ERROR", "CRITICAL"],
            help="""
                Set the log level.
            """
        )

Reply:
1. You're right, that's from code I was working on before. Will fix this.

2. Something about this implementation doesn't catch the error correctly without an explicit validation. With my current implementation we get the error:
```shell
[rank1]:  File "/mnt/sxtn/phil/YNNtitan/torchtitan/logging.py", line 43, in from_string
[rank1]:    raise argparse.ArgumentTypeError(f"Invalid log level: '{value}'. "
[rank1]:argparse.ArgumentTypeError: Invalid log level: 'aKD'. Choose from DEBUG, INFO, WARNING, ERROR, CRITICAL.
W0903 22:12:28.782000 1660880 torch/distributed/elastic/multiprocessing/api.py:890] Sending process 1660907 closing signal SIGTERM
```

If we use choices, we get the following stack trace

Reply:
May be there is some kind of post-processing, because by deafult it should raise a parser error like this
 
![Screenshot from 2024-09-04 13-01-57](https://github.com/user-attachments/assets/bc83f687-82fa-4bb0-bb19-bd2c0503c639)

Reply:
OK, let's move on with your approach.

Reply:
Just want to clarify for future reference, that this occurs because the torchtitan code is not parsing with parse args to allow for more flexibility:
https://github.com/YerevaNN/YNNtitan/blob/a6c54844f690890c37418cfdbb519889707333a0/torchtitan/config_manager.py#L570

For this reason, we don't have choice checking and type adherence from argparse.